torch/csrc/distributed/c10d/c10d_frontend.h

```diff
@@ -186,33 +186,35 @@ class DistributedC10d {
       std::chrono::milliseconds timeout,
       Backend backend);
 
+  std::shared_ptr<ProcessGroup> worldProcessGroup();
+
  private:
   DistributedC10d(){};
 
-  bool rankNotInGroup(std::shared_ptr<ProcessGroup> group) const;
-  int64_t getGroupRank(std::shared_ptr<ProcessGroup> group, const int64_t rank)
+  bool rankNotInGroup(const std::shared_ptr<ProcessGroup>& group) const;
+  int64_t getGroupRank(const std::shared_ptr<ProcessGroup> group&, const int64_t rank)
       const;
   int64_t getGlobalRank(
-      std::shared_ptr<ProcessGroup> group,
+      const std::shared_ptr<ProcessGroup>& group,
       const int64_t global_rank) const;
   void checkDefaultPg() const;
-  int64_t getGroupSize(std::shared_ptr<ProcessGroup> group) const;
-  int64_t getBackend(std::shared_ptr<ProcessGroup> group);
+  int64_t getGroupSize(const std::shared_ptr<ProcessGroup>& group) const;
+  std::string getBackend(const std::shared_ptr<ProcessGroup>& group);
 
   std::string backend_;
   // TODO: Ask Alex what kind of equality we need. It determine whether we
   // need to use ProcessGroup or ProcesGroup* as key.
   std::unordered_map<
       std::shared_ptr<ProcessGroup>,
-      std::pair<std::shared_ptr<Backend>, std::shared_ptr<Store>>>
+      std::pair<std::string, std::shared_ptr<Store>>>
       pg_map_;
 
   // Note, this is different mapping relationship than original Python
   // implementation.
   std::unordered_map<std::shared_ptr<ProcessGroup>, std::string> pg_names_;
 
   // Value is global_rank:group_rank mapping.
-  std::unordered_map<std::shared_ptr<ProcessGroup>, std::vector<int64_t>>
+  std::unordered_map<std::shared_ptr<ProcessGroup>, std::unordered_map<int64_t, int64_t>>
```

Comment:
can you document what's the inner map semantics?

Reply:
Copied over Python layer's comment.